saos-batch/src/test/java/pl/edu/icm/saos/batch/core/importer/CtJudgmentImportJobTest.java

```diff
@@ -0,0 +1,243 @@
+    
+    //------------------------ TESTS --------------------------
+    
+    @Test
+    public void ctJudgmentImportProcessJob_IMPORT_NEW() throws Exception {
+        
+        resetImportDir("import/constitutionalTribunal/judgments/version1");
+        
+        JobExecution jobExecution = jobExecutor.forceStartNewJob(ctJudgmentImportJob);
+        
+        assertEquals(6, rJudgmentRepository.count(RawSourceCtJudgment.class));
+        assertEquals(0, rJudgmentRepository.findAllNotProcessedIds(RawSourceCtJudgment.class).size());
+        JobExecutionAssertUtils.assertJobExecution(jobExecution, 0, 6);
+        
+        assertEquals(6, judgmentRepository.count(ConstitutionalTribunalJudgment.class));
+        assertSourceJudgmentIds("3b42a6299303c65d869c4806fdcdbf7a", "5a7ec04c9f5d354e00027929ed86025a",
+                "12f3b546205345a265acf9a39c491c6a", "0e8b967bd3c71e3eec89630d5baee5e1",
+                "04e47013023d2b315b841f99ccb9c290", "281c0d5a6739a1e754c2b7b56effb1b6");
+        
+        assertJudgment_3b42a6299303c65d869c4806fdcdbf7a();
+        
+        
+    }
+    
+    @Test
+    public void ctJudgmentImportProcessJob_IMPORT_UPDATE() throws Exception {
+        
+        // given
+        
+        resetImportDir("import/constitutionalTribunal/judgments/version1");
+        jobExecutor.forceStartNewJob(ctJudgmentImportJob);
+        int ctJudgment04e47013Id = judgmentRepository.findOneBySourceCodeAndSourceJudgmentId(SourceCode.CONSTITUTIONAL_TRIBUNAL, "04e47013023d2b315b841f99ccb9c290").getId();
+        
+        resetImportDir("import/constitutionalTribunal/judgments/version2");
+        
+        
+        // execute
+        
+        JobExecution jobExecution = jobExecutor.forceStartNewJob(ctJudgmentImportJob);
+        
+        
+        // assert
+        
+        assertEquals(5, rJudgmentRepository.count(RawSourceCtJudgment.class));
+        assertEquals(0, rJudgmentRepository.findAllNotProcessedIds(RawSourceCtJudgment.class).size());
+        JobExecutionAssertUtils.assertJobExecution(jobExecution, 0, 5);
+        
+        assertEquals(5, judgmentRepository.count(ConstitutionalTribunalJudgment.class));
+        assertSourceJudgmentIds("3b42a6299303c65d869c4806fdcdbf7a", "5a7ec04c9f5d354e00027929ed86025a",
+                "12f3b546205345a265acf9a39c491c6a", "0e8b967bd3c71e3eec89630d5baee5e1",
+                "04e47013023d2b315b841f99ccb9c290");
+        
+        assertNull(judgmentRepository.findOneBySourceCodeAndSourceJudgmentId(
+                SourceCode.CONSTITUTIONAL_TRIBUNAL, "281c0d5a6739a1e754c2b7b56effb1b6", ConstitutionalTribunalJudgment.class));
+        
+        
+        assertEquals(ctJudgment04e47013Id, judgmentRepository.findOneBySourceCodeAndSourceJudgmentId(SourceCode.CONSTITUTIONAL_TRIBUNAL, "04e47013023d2b315b841f99ccb9c290").getId());
+        
+        assertJudgment_04e47013023d2b315b841f99ccb9c290_afterUpdate();
+        assertCorrections_04e47013023d2b315b841f99ccb9c290_afterUpdate();
```

Comment:
In ctJudgmentImportProcessJob_IMPORT_NEW and ctJudgmentImportProcessJob_IMPORT_UPDATE you check different judgments. This way it's hard to say if anything has changed during update. In my view you should additionaly check in ctJudgmentImportProcessJob_IMPORT_NEW the state of 04e before update.